### PR TITLE
chore(debugger): compile expression lambda with access locals variables

### PR DESCRIPTION
**Pull request by shatzi**
## Description
Debugger expression lambda should be able to access local variables. so user can generate expression like:
`any(array, { @it.value = #arg1 && @it.id = ^local2 } )`

## Motivation
Align with dd-trace-java debugger features

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: ddtrace/debugging/_expressions.py [hunk 1]
```diff
@@ -132,17 +149,14 @@ def _compile_arg_predicate(ast):
...
-            + ca
-            + [
-                Instr("LOAD_CONST", fb),
-                Instr("CALL_FUNCTION", 2),
-            ]
-        )
+        if cb is None:
+            raise ValueError("Invalid lmabda: %r" % b)
+
+        return [Instr("LOAD_CONST", lambda i, c: f(c(_) for _ in i))] + ca + cb + [Instr("CALL_FUNCTION", 2)]
```
Comment (P403n1x87) [inline ddtrace/debugging/_expressions.py:159]: I think we could probably simplify this slightly by doing something like
```suggestion
        return [Instr("LOAD_CONST", lambda i, c, _locals: f(c(_, _locals) for _ in i))] + ca + cb + [Instr("CALL_FUNCTION", 2)]
```
and the pushing a third argument to the stack, before `CALL_FUNCTION` with
```
Instr("LOAD_FAST", "_locals")
```
Then I think we shouldn't need the new helpers. WDYT?

File: tests/submod/stuff.py [hunk 2]
```diff
@@ -135,3 +135,7 @@ def durationstuff(ns):
 
 def mutator(arg):
     arg.append(42)
+
+
+def ageChecker(people, age, name=None):
```
Comment (P403n1x87) [inline tests/submod/stuff.py:140]: snake case 🙂 
```suggestion
def age_checker(people, age, name=None):
```

File: tests/debugging/test_debugger.py [hunk 3]
```diff
@@ -742,3 +742,31 @@ def test_debugger_function_probe_eval_on_exit():
 
         (snapshot,) = d.test_queue
         assert snapshot, d.test_queue
+
+
+def test_debugger_lamba_fuction_access_locals():
```
Comment (P403n1x87) [inline tests/debugging/test_debugger.py:747]:
```suggestion
def test_debugger_lambda_fuction_access_locals():
```

**Review by P403n1x87 — APPROVED** [review]
🙌

**Review by mabdinur — APPROVED** [review]

**Review by mabdinur — COMMENTED** [review]
This appears to be a new feature. Can we add a releasenote? Otherwise LGTM

**Comment by P403n1x87** [comment]
> This appears to be a new feature. Can we add a releasenote? Otherwise LGTM

As with other small enhancements in this area, I think we can get away with no relno for now, and reduce the verbosity of the changelog, until these code paths will actually be exercised by users.

**Pull request merged**